woocommerce/woocommerce-android
# [Simplified login] Refactor PluginRepository to prepare for native Jetpack Installation

Closes: #7795 

⚠️ Depends on https://github.com/wordpress-mobile/WordPress-FluxC-Android/pull/2565

### Description
This PR refactors the `PluginRepository` to make it reusable for both Jetpack sites and non-Jetpack sites, this will allow us to add support for installing Jetpack natively during the login flow.

### Testing instructions
The work to use the repository using the WPApi and site credentials will be added soon, for now, we should focus on non-regression tests:
1. Create a JCP site (please use Jetpack Backup as plugin, WCPay uses an outdated version of the JCP library p1668187859711119-slack-CPBTESZMG)
2. Open the app, then sign to this site.
3. Click on the bottom banner to start Jetpack installation.
4. Proceed to the installation, and confirm it works as expected.
